pysindy/feature_library/base.py

```diff
@@ -99,7 +159,7 @@ def _ensemble(self, xp):
                 )
             inds = range(self.n_output_features_)
             inds = np.delete(inds, self.ensemble_indices)
-            return xp[:, inds]
+            return [x[..., inds] for x in xp]
```

Comment:
The ellipses `...` are used in many libraries to slice only along the `ax_coords` axis. For future usage where the axes aren't necessarily ordered with `ax_coord` at the end, it would be nice to include a utility function to slice along the `ax_coord`.

Reply:
`x.take(inds, axis=x.ax_coord)`

Reply:
Also, not sure this will work for assignment of slices along the `ax_coord` axis, like `spatiotemporal_grid[..., -1] = temporal_grid`. Would be easy to implement with a slice object, like 
```
s=[slice(None)]*spatiotemporal_grid.ndim
s[spatiotemporal_grid.ax_coord]=-1
spatiotemporal_grid[tuple(s)] = -1
```
Not sure whether a utility function in AxesArray would simplify code like this or not, but maybe.

Reply:
I haven't used it, it looks like `np.put_along_axis` seems to be the counterpart to `np.take`

Reply:
Cool, good to know, thanks!